File: src/main/java/mods/eln/Achievements.java
package mods.eln;

import mods.eln.registry.RegistryUtils;
import net.minecraft.stats.Achievement;
import net.minecraftforge.common.AchievementPage;

import static mods.eln.i18n.I18N.*;

public class Achievements {

    public static Achievement craft50VMacerator;
    public static AchievementPage achievementPageEln;

    public static void init() {
        // the last null is the parent achievement. TODO: Make actual achievements system :)
        craft50VMacerator = new Achievement(TR("achievement.craft_50v_macerator"),
            "craft_50v_macerator", 0, 2, RegistryUtils.findItemStack("50V Macerator", 0), null).registerStat();

        TR_DESC(Type.ACHIEVEMENT, "craft_50v_macerator");

        achievementPageEln = new AchievementPage(tr("Electrical Age [WIP]"),
            craft50VMacerator);

        AchievementPage.registerAchievementPage(achievementPageEln);
    }
}

File: src/main/kotlin/mods/eln/Achievements.kt
package mods.eln

import mods.eln.i18n.I18N
import mods.eln.i18n.I18N.tr
import net.minecraft.init.Items
import net.minecraft.stats.Achievement
import net.minecraftforge.common.AchievementPage

object Achievements {
    @JvmField
    var openGuide: Achievement? = null
    @JvmField
    var craft50VMacerator: Achievement? = null
    var achievementPageEln: AchievementPage? = null

    @JvmStatic
    fun init() {
        openGuide = Achievement(
            tr("achievement.open_guide"),
            "open_guide", 0, 0, Items.book, null
        ).registerStat()

        I18N.TR_DESC(I18N.Type.ACHIEVEMENT, "open_guide")

        craft50VMacerator = Achievement(
            tr("achievement.craft_50v_macerator"),
            "craft_50v_macerator", 0, 2, Eln.findItemStack("50V Macerator", 0), openGuide
        ).registerStat()

        I18N.TR_DESC(I18N.Type.ACHIEVEMENT, "craft_50v_macerator")

        achievementPageEln = AchievementPage(
            tr("Electrical Age [WIP]"),
            openGuide, craft50VMacerator
        )

        AchievementPage.registerAchievementPage(achievementPageEln)
    }
}
